tests/test_torch_compile.py

```
@@ -0,0 +1,105 @@
+    return attention_config, norm_config
+
+
+@pytest.fixture
+def gpt2_model():
+    attention_config, norm_config = create_gpt2_configs()
+    model = GPT2LLM(
+        sample_key="input_ids",
+        prediction_key="logits",
+        poe_type=PositionTypes.NOPE,
+        sequence_length=256,
+        vocab_size=1024,
+        n_layer=4,
+        n_head_q=8,
+        n_head_kv=4,
+        n_embd=512,
+        ffn_hidden=2048,
+        dropout=0.1,
+        bias=True,
+        activation_type=ActivationType.SWIGLU,
+        attention_implementation=AttentionImplementation.PYTORCH_FLASH,
+        attention_config=attention_config,
+        attention_norm_config=norm_config,
+        ffn_norm_config=norm_config,
+        lm_head_norm_config=norm_config,
+        use_weight_tying=False,
+    )
+    return model
+
+
+def test_get_compiled_model_compiles_blocks(gpt2_model):
+    original_blocks = list(gpt2_model.transformer.h)
+    original_wte = gpt2_model.transformer.wte
+    original_lm_head = gpt2_model.transformer.lm_head
+
+    block_names = ["GPT2Block"]
+    result_model = ModelFactory.get_compiled_model(gpt2_model, block_names)
+
+    assert len(result_model.transformer.h) == 4, "Should still have four blocks"
+    for i, (original_block, new_block) in enumerate(zip(original_blocks, result_model.transformer.h)):
+        assert new_block is not original_block, f"Block {i} should be a compiled version"
+        assert isinstance(new_block, nn.Module), f"Block {i} should be an nn.Module"
+    assert result_model.transformer.wte is original_wte, "Embedding layer should remain unchanged"
+    assert result_model.transformer.lm_head is original_lm_head, "LM head should remain unchanged"
+    assert result_model is gpt2_model, "Should return the same model instance"
+
+
+def test_get_compiled_model_invalid_block_name(gpt2_model):
+    """
+    Test that get_compiled_model does nothing when given an invalid block name.
```

I think we should raise an Exception in this case.

---

Raises now an exception:
https://github.com/Modalities/modalities/pull/344/commits/bf60a361c55cdb25ba21aeb4904cea032dc83acb